bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -0,0 +1,307 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.openthermgateway.handler;
+
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.openthermgateway.OpenThermGatewayBindingConstants;
+import org.openhab.binding.openthermgateway.internal.DataItem;
+import org.openhab.binding.openthermgateway.internal.DataItemGroup;
+import org.openhab.binding.openthermgateway.internal.GatewayCommand;
+import org.openhab.binding.openthermgateway.internal.GatewayCommandCode;
+import org.openhab.binding.openthermgateway.internal.LogLevel;
+import org.openhab.binding.openthermgateway.internal.Message;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayCallback;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayConfiguration;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayConnector;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewaySocketConnector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenThermGatewayHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Arjen Korevaar - Initial contribution
+ */
+@NonNullByDefault
+public class OpenThermGatewayHandler extends BaseThingHandler implements OpenThermGatewayCallback {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenThermGatewayHandler.class);
+
+    @Nullable
+    private OpenThermGatewayConfiguration config;
+
+    @Nullable
+    private OpenThermGatewayConnector connector;
+
+    public OpenThermGatewayHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing OpenTherm Gateway handler for uid '{}'", getThing().getUID());
+
+        updateStatus(ThingStatus.OFFLINE);
+
+        config = getConfigAs(OpenThermGatewayConfiguration.class);
+
+        connect();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Received channel: {}, command: {}", channelUID, command);
+
+        try {
+            if (command.toFullString() != "REFRESH") {
+                String channel = channelUID.getId();
+                String code = getGatewayCodeFromChannel(channel);
+
+                GatewayCommand gatewayCommand;
+                if (command instanceof QuantityType) {
```

Comment:
Currently there are no channels that have a QuantityType so this check is not needed. However I would suggest to make some of the channels with unit UoM (see my other comment). In that case you can do this check. However you need to pass the correct unit to get the value of the command otherwise you might end up with a different unit you expect. For example if it's a temperature unit, but the command contains Fahrenheit you need to get the Celsius value probably. So you need to pass Celcius to get the double value.

Reply:
Changed the temperature channels to UoM back and forth